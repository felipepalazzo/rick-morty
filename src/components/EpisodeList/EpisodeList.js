import React from 'react'
import PropTypes from 'prop-types'

export default function EpisodeList({ episodes }) {
	return (
		<section>
			<div>
				<strong>Episodes: </strong>
			</div>
			{episodes.map((ep, idx) => (
				<span key={ep.id}>
					{ep.name}
					{idx !== episodes.length - 1 ? ' / ' : ''}
				</span>
			))}
		</section>
	)
}

EpisodeList.defaultProps = {
	episodes: []
}

EpisodeList.propTypes = {
	episodes: PropTypes.array
}
