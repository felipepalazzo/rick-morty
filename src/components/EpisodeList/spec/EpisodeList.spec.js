import React from 'react'
import { shallow } from 'enzyme'
import EpisodeList from '../EpisodeList'

describe('EpisodeList', () => {
	const props = {
		episodes: [{ id: 1, name: 'name1' }, { id: 2, name: 'name2' }]
	}
	const wrapper = shallow(<EpisodeList {...props} />)
	it('renders without crashing', () => {
		expect(wrapper).toMatchSnapshot()
	})
	it('should render the correct num of items', () => {
		expect(wrapper.find('span')).toHaveLength(2)
	})
})
