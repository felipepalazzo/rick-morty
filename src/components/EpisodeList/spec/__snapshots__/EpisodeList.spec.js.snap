// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeList renders without crashing 1`] = `ShallowWrapper {}`;
